File: frontend-flutter/lib/models/livraison.dart
import 'commande.dart';
import 'operateur.dart';

class Livraison {
  final int idLivraison;
  final Commande commande;
  final String status;
  final String qrToken;
  final DateTime? createdAt;
  final DateTime? updatedAt;
  final Operateur? lastUpdatedBy;

  Livraison({
    required this.idLivraison,
    required this.commande,
    required this.status,
    required this.qrToken,
    this.createdAt,
    this.updatedAt,
    this.lastUpdatedBy,
  });

  factory Livraison.fromJson(Map<String, dynamic> json) {
    return Livraison(
      idLivraison: (json['idLivraison'] ?? 0) as int,
      commande: Commande.fromJson((json['commande'] ?? {}) as Map<String, dynamic>),
      status: (json['status'] ?? '') as String,
      qrToken: (json['qrToken'] ?? '') as String,
      createdAt: json['createdAt'] != null ? DateTime.tryParse(json['createdAt']) : null,
      updatedAt: json['updatedAt'] != null ? DateTime.tryParse(json['updatedAt']) : null,
      lastUpdatedBy: json['lastUpdatedBy'] != null
          ? Operateur.fromJson((json['lastUpdatedBy']) as Map<String, dynamic>)
          : null,
    );
  }
}

File: backend-springboot/src/main/java/com/esprit/examen/entities/Livraison.java
package com.esprit.examen.entities;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.time.Instant;
import java.util.UUID;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Livraison implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idLivraison;

    @OneToOne(optional = false)
    private Commande commande;

    @Enumerated(EnumType.STRING)
    private LivraisonStatus status = LivraisonStatus.EN_ATTENTE;

    /**
     * Token used inside the QR code. We avoid exposing sequential IDs.
     */
    @Column(unique = true, nullable = false, updatable = false)
    private String qrToken = UUID.randomUUID().toString();

    private Instant createdAt = Instant.now();
    private Instant updatedAt = Instant.now();

    @ManyToOne
    private Operateur lastUpdatedBy;
}
